File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/monitoring/port/StagePerformanceRollupRepository.java
package com.study.webflux.rag.domain.monitoring.port;

import java.time.Instant;

import com.study.webflux.rag.domain.monitoring.model.MetricsGranularity;
import com.study.webflux.rag.domain.monitoring.model.StagePerformanceRollup;
import reactor.core.publisher.Flux;

public interface StagePerformanceRollupRepository {
	Flux<StagePerformanceRollup> saveAll(Flux<StagePerformanceRollup> rollups);

	Flux<StagePerformanceRollup> findByGranularityAndBucketStartBetween(
		MetricsGranularity granularity,
		Instant startTime,
		Instant endTime);
}

File: domain/src/main/kotlin/com/miyou/app/domain/monitoring/port/StagePerformanceRollupRepository.kt
package com.miyou.app.domain.monitoring.port

import com.miyou.app.domain.monitoring.model.MetricsGranularity
import com.miyou.app.domain.monitoring.model.StagePerformanceRollup
import reactor.core.publisher.Flux
import java.time.Instant

interface StagePerformanceRollupRepository {
    fun saveAll(rollups: Flux<StagePerformanceRollup>): Flux<StagePerformanceRollup>

    fun findByGranularityAndBucketStartBetween(
        granularity: MetricsGranularity,
        startTime: Instant,
        endTime: Instant,
    ): Flux<StagePerformanceRollup>
}
